kii-core/kii_core.c

```diff
@@ -384,8 +384,9 @@ prv_kii_http_append_body(
         return KII_HTTPC_FAIL;
     }
 
-    strncat(http_context->buffer, body, body_len);
-    http_context->total_send_size = kii_strlen(http_context->buffer);
+    memcpy(http_context->buffer + http_context->total_send_size,
```

Comment:
How about using memmove?
Although the memory area would not overlap, 'undefined' behavior is scary when it is not works as intended.